shark-graph/src/test/java/shark/HprofDeobfuscatorTest.kt

```diff
@@ -0,0 +1,148 @@
+package shark
+
+import org.assertj.core.api.Assertions
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TemporaryFolder
+import shark.HprofRecord.HeapDumpRecord.ObjectRecord.ClassDumpRecord
+import shark.HprofRecord.HeapDumpRecord.ObjectRecord.ClassDumpRecord.FieldRecord
+import shark.HprofRecord.HeapDumpRecord.ObjectRecord.ClassDumpRecord.StaticFieldRecord
+import shark.HprofRecord.HeapDumpRecord.ObjectRecord.InstanceDumpRecord
+import shark.HprofRecord.LoadClassRecord
+import shark.HprofRecord.StringRecord
+import shark.ValueHolder.ReferenceHolder
+import java.io.File
+
+class HprofDeobfuscatorTest {
+
+  @get:Rule
+  var testFolder = TemporaryFolder()
+
+  private var lastId = 0L
+  private val id: Long
+    get() = ++lastId
+
+  @Test
+  fun deobfuscateHprof() {
```

Comment:
Can you separate this out in several tests?

1) Class name mapping
2) Member field mapping
3) Static field mapping

Then can you add tests for edge cases (I know it works right now but should help with future changes):

* class name is also used as member field name
* class name is also used as static field name
* Two fields with different names in two distinct classes are obfuscated to the same field name

Any other edge case I forgot?

Reply:
I separated those tests out.